app/livechat/server/startup.js

```diff
@@ -11,6 +11,27 @@ import { RoutingManager } from './lib/RoutingManager';
 import { createLivechatQueueView } from './lib/Helper';
 import { LivechatAgentActivityMonitor } from './statistics/LivechatAgentActivityMonitor';
 
+function allowAccessClosedRoomOfSameDepartment(room, user) {
+	if (!room || !user || room.t !== 'l' || !room.departmentId || room.open) {
+		return;
+	}
+	const agentOfDepartment = LivechatDepartmentAgents.findOneByAgentIdAndDepartmentId(user._id, room.departmentId);
+	if (!agentOfDepartment) {
+		return;
+	}
+	return hasPermission(user._id, 'view-livechat-closed-room-same-department');
+}
+
+function allowAccessToClosedRoomsByAnotherAgent(room, user) {
```

Comment:
I don't think we need another method for this case, I think we can use an `addRoomAccessValidator` that we already have, this one: 
https://github.com/RocketChat/Rocket.Chat/blob/develop/app/livechat/server/startup.js#L21

The only change to do is on the return statement:
From:
```
    return userId === agentId;
```

What do you think?

To:
```
    return userId === agentId || (!room.open && hasPermission(user._id, 'view-livechat-closed-room-by-another-agent');
```